recipes/tensorflow/samples/tensorflow/retail/retail/tensorflow/trainer.py

```diff
@@ -0,0 +1,156 @@
+#####################################################################
+# ADOBE CONFIDENTIAL
+# ___________________
+#
+#  Copyright 2019 Adobe
+#  All Rights Reserved.
+#
+# NOTICE:  All information contained herein is, and remains
+# the property of Adobe and its suppliers, if any. The intellectual
+# and technical concepts contained herein are proprietary to Adobe
+# and its suppliers and are protected by all applicable intellectual
+# property laws, including trade secret and copyright laws.
+# Dissemination of this information or reproduction of this material
+# is strictly forbidden unless prior written permission is obtained
+# from Adobe.
+#####################################################################
+
+from ml.runtime.python.Interfaces.AbstractTrainer import AbstractTrainer
+from data_access_sdk_python.reader import DataSetReader
+import numpy as np
+import pandas as pd
+
+import tensorflow as tf
+fc = tf.feature_column
+tf.set_random_seed(123)
+
+import os
+import json
+import pickle
+
+class Trainer(AbstractTrainer):
+    def train(self, config={}):
+        #########################################
+        # Set Up
+        #########################################
+        tf.logging.set_verbosity(tf.logging.ERROR)
+        tf_config = json.loads(os.environ['TF_CONFIG'])
+        tf_config = json.loads('{}')
+        os.environ['TF_CONFIG'] = json.dumps(tf_config)
+
+
+
+        #########################################
+        # Load Data
+        #########################################
+        prodreader = DataSetReader(client_id=config['ML_FRAMEWORK_IMS_USER_CLIENT_ID'],
+                                   user_token=config['ML_FRAMEWORK_IMS_TOKEN'],
+                                   service_token=config['ML_FRAMEWORK_IMS_ML_TOKEN'])
+
+        dataframe = prodreader.load(data_set_id=config['trainingDataSetId'],
+                                    ims_org=config['ML_FRAMEWORK_IMS_ORG_ID'])
+
+
+
+        #########################################
+        # Data Preparation/Feature Engineering
+        #########################################
+        if '_id' in dataframe.columns:
+            # Rename columns to strip tenantId
+            dataframe = dataframe.rename(columns=lambda x: str(x)[str(x).find('.') + 1:])
+            # Drop id, eventType and timestamp
+            dataframe.drop(['_id', 'eventType', 'timestamp'], axis=1, inplace=True)
```

Comment:
Repeated code here - good candidate for refactoring at some point.

Reply:
Yes, certainly.